packages/connectors/connector-wecom/src/index.ts

```diff
@@ -0,0 +1,199 @@
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, wecomConfigGuard);
+
+    // WeCom has two different authorizationEndpoint, one for the WeCom built-in browser and another for regular web browsers.
+    const isWecom: boolean | undefined = userAgent?.toLowerCase().includes('wxwork');
+    const { corpId, scope, agentId } = config;
+
+    const queryParameters = new URLSearchParams({
+      appid: corpId,
+      redirect_uri: encodeURI(redirectUri), // The variable `redirectUri` should match {appId, appSecret}
+      response_type: 'code',
+      scope: scope ?? defaultScope,
+      state,
+      agentid: agentId,
+    });
+    // We switch to different authorizationEndpoint based on the value of isWecom
+    const authorizationEndpoint = isWecom
+      ? authorizationEndpointInside
+      : authorizationEndpointQrcode;
+
+    return `${authorizationEndpoint}?${queryParameters.toString()}#wechat_redirect`;
+  };
+
+export const getAccessToken = async (
+  config: WecomConfig
+): Promise<{ accessToken: string }> => {
+  const { corpId: corpid, appSecret: corpsecret } = config;
+
+  const httpResponse = await got.get(accessTokenEndpoint, {
+    searchParams: { corpid, corpsecret },
+    timeout: { request: defaultTimeout },
+  });
+
+  const result = accessTokenResponseGuard.safeParse(parseJson(httpResponse.body));
+
+  if (!result.success) {
+    throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+  }
+
+  const { access_token: accessToken } = result.data;
+
+  getAccessTokenErrorHandler(result.data);
+
+  assert(accessToken, new ConnectorError(ConnectorErrorCodes.InvalidResponse));
+
+  return { accessToken };
+};
+
+const getUserInfo =
+  (getConfig: GetConnectorConfig): GetUserInfo =>
+  async (data) => {
+    const { code } = await authorizationCallbackHandler(data);
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, wecomConfigGuard);
+    const { accessToken } = await getAccessToken(config);
+
+    try {
+      const httpResponse = await got.get(userInfoEndpoint, {
+        searchParams: { access_token: accessToken, code },
+        timeout: { request: defaultTimeout },
+      });
+
+      const result = userInfoResponseGuard.safeParse(parseJson(httpResponse.body));
+
+      if (!result.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, result.error);
+      }
+
+      const { userid, openid } = result.data;
+
+      // Response properties of user info can be separated into two groups: (1) {unionid, headimgurl, nickname}, (2) {errcode, errmsg}.
+      // These two groups are mutually exclusive: if group (1) is not empty, group (2) should be empty and vice versa.
+      // 'errmsg' and 'errcode' turn to non-empty values or empty values at the same time. Hence, if 'errmsg' is non-empty then 'errcode' should be non-empty.
+      userInfoResponseMessageParser(result.data);
```

Comment:
Like the detailed comments here :D
This method can be renamed as something like `errorResponseHandler`. By looking into its inner logic, you can put this method before destructuring `result.data`, since the right response with user info and error response are mutually exclusive and when the error is met then there is no need to destruct `result.data`. What do you think?